src/backend/distributed/commands/schema.c

```diff
@@ -61,11 +118,6 @@ PreprocessDropSchemaStmt(Node *node, const char *queryString,
 	Oid scanIndexId = InvalidOid;
 	bool useIndex = false;
 
-	if (dropStatement->behavior != DROP_CASCADE)
-	{
-		return NIL;
-	}
-
 	Value *schemaValue = NULL;
 	foreach_ptr(schemaValue, dropStatement->objects)
```

Comment:
I know that this for-loop predates your changes, I have a hard time understanding what we are doing here.

If you have figured out what this loop is for can you do any of the following:
 - leave a comment on top of the loop explaining its purpose
 - extract the loop into a self describing function

If you are not completely sure yourself,  please ignore this comment. It is not required for your change, mostly trying to leave the camping site better then we found it now that we are here.